### ISSUE TITLE
WARNING: GPU found but will not be used. Please set `experiment.config.trial_gpu_number` to the number of GPUs you want to use for each trial.

### ISSUE THREAD

**Issue by yyh7773** [microsoft/nni#5708]
''Hello,NAS'':GPU can't be used but i set all:
and if i use CPU,there is a error :策略执行失败

**Environment**:
- NNI version:3.0
- Training service (local|remote|pai|aml|etc):；local
- Client OS:Windows 10
- Server OS (for remote mode only):No
- Python version:3.10
- PyTorch/TensorFlow version:1.12.1+cu113
- Is conda/virtualenv/venv used?:Pycharm Edit
- Is running in Docker?:No

**Configuration**:
 - Experiment config (remember to remove secrets!):{
  "experimentType": "nas",
  "searchSpace": {},
  "trialCommand": "D:\\learnsaoftware\\anaconda3\\python.exe -m nni.nas.execution.training_service trial",
  "trialCodeDirectory": "C:\\Users\\Administrator\\Desktop\\yyh",
  "trialConcurrency": 100,
  "trialGpuNumber": 1,
  "maxTrialNumber": 100,
  "useAnnotation": false,
  "debug": false,
  "logLevel": "info",
  "experimentWorkingDirectory": "C:\\Users\\Administrator\\nni-experiments",
  "trainingService": {
    "platform": "local",
    "trialCommand": "D:\\learnsaoftware\\anaconda3\\python.exe -m nni.nas.execution.training_service trial",
    "trialCodeDirectory": "C:\\Users\\Administrator\\Desktop\\yyh",
    "trialGpuNumber": 1,
    "debug": false,
    "useActiveGpu": true,
    "maxTrialNumberPerGpu": 1,
    "reuseMode": false
  },
  "executionEngine": {
    "name": "ts"
  },
  "modelFormat": {
    "name": "simplified"
  }
}
 - Search space:class MyModelSpace(ModelSpace):
    def __init__(self):
        super().__init__()
        self.conv1 = nn.Conv2d(1, 32, 3, 1)
        # LayerChoice is used to select a layer between Conv2d and DwConv.
        self.conv2 = LayerChoice([
            nn.Conv2d(32, 64, 3, 1),
            DepthwiseSeparableConv(32, 64)
        ], label='conv2')
        # nni.choice is used to select a dropout rate.
        # The result can be used as parameters of `MutableXXX`.
        self.dropout1 = MutableDropout(nni.choice('dropout', [0.25, 0.5, 0.75]))  # choose dropout rate from 0.25, 0.5 and 0.75
        self.dropout2 = nn.Dropout(0.5)
        feature = nni.choice('feature', [64, 128, 256])
        self.fc1 = MutableLinear(9216, feature)
        self.fc2 = MutableLinear(feature, 10)

    def forward(self, x):
        x = F.relu(self.conv1(x))
        x = F.max_pool2d(self.conv2(x), 2)
        x = torch.flatten(self.dropout1(x), 1)
        x = self.fc2(self.dropout2(F.relu(self.fc1(x))))
        output = F.log_softmax(x, dim=1)
        return output

model_space = MyModelSpace()

上述是我的设置情况，请问应该如何解决？

**Comment by yyh7773** [comment]
MyModelSpace(
  (conv1): Conv2d(1, 32, kernel_size=(3, 3), stride=(1, 1))
  (conv2): LayerChoice(
    label='conv2'
    (0): Conv2d(32, 64, kernel_size=(3, 3), stride=(1, 1))
    (1): DepthwiseSeparableConv(
      (depthwise): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), groups=32)
      (pointwise): Conv2d(32, 64, kernel_size=(1, 1), stride=(1, 1))
    )
  )
  (dropout1): MutableDropout(p=Categorical([0.25, 0.5, 0.75], label='dropout'))
  (dropout2): Dropout(p=0.5, inplace=False)
  (fc1): MutableLinear(in_features=9216, out_features=Categorical([64, 128, 256], label='feature'))
  (fc2): MutableLinear(in_features=Categorical([64, 128, 256], label='feature'), out_features=10)
)
我的搜索空间与Demo中一样

**Comment by XuanmaiXue** [comment]
I got same problem, have you solved it ?
